Netable/Netable/Netable.swift

```diff
@@ -170,13 +189,38 @@ public actor Netable {
         return (task: task, subject: resultSubject.receive(on: RunLoop.main))
     }
 
-    private func startRequestTask<T: Request>(_ request: T, urlRequest: URLRequest, id: String) async throws -> T.FinalResource {
+    private func startRequestTask<T: Request>(
+        _ request: T,
+        interceptor: Interceptor?,
+        urlRequest: URLRequest,
+        id: String
+    ) async throws -> T.FinalResource {
         let startTimestamp = CACurrentMediaTime()
 
+        var finalUrlRequest = urlRequest
+        var mockedUrl: URL?
+        if let interceptor = self.interceptor {
+            let result = try await interceptor.applyInterceptors(request: finalUrlRequest, instance: self)
+            switch result {
+            case .changed(let newRequest): finalUrlRequest = newRequest
+            case .mocked(let url): mockedUrl = url
+            case .notChanged: break
+            }
+        }
+
+        if let interceptor = interceptor {
+            let result = try await interceptor.applyInterceptors(request: finalUrlRequest, instance: self)
+            switch result {
+            case .changed(let newRequest): finalUrlRequest = newRequest
+            case .mocked(let url): mockedUrl = url
+            case .notChanged: break
+            }
+        }
```

Comment:
Don't love that we've got two copies of the exact same code back to back here, seems like there's gotta be some way to refactor this so we can like merge the two lists and just run apply once.